packages/expo-camera/android/src/main/java/expo/modules/camera/CameraModule.kt

```diff
@@ -2,216 +2,185 @@ package expo.modules.camera
 
 import android.Manifest
 import android.content.Context
-
 import com.google.android.cameraview.AspectRatio
-
 import expo.modules.camera.tasks.ResolveTakenPictureAsyncTask
-import expo.modules.core.ExportedModule
-import expo.modules.core.ModuleRegistry
-import expo.modules.core.interfaces.ExpoMethod
-import expo.modules.core.ModuleRegistryDelegate
-import expo.modules.core.Promise
-import expo.modules.core.interfaces.services.UIManager
 import expo.modules.core.utilities.EmulatorUtilities
 import expo.modules.interfaces.permissions.Permissions
+import expo.modules.kotlin.Promise
+import expo.modules.kotlin.exception.Exceptions
+import expo.modules.kotlin.functions.Queues
+import expo.modules.kotlin.modules.Module
+import expo.modules.kotlin.modules.ModuleDefinition
+
+class CameraModule : Module() {
+  override fun definition() = ModuleDefinition {
+    Name("ExponentCameraModule")
+    Constants(
+      "Type" to mapOf(
+        "front" to com.google.android.cameraview.Constants.FACING_FRONT,
+        "back" to com.google.android.cameraview.Constants.FACING_BACK
+      ),
+      "FlashMode" to mapOf(
+        "off" to com.google.android.cameraview.Constants.FLASH_OFF,
+        "on" to com.google.android.cameraview.Constants.FLASH_ON,
+        "auto" to com.google.android.cameraview.Constants.FLASH_AUTO,
+        "torch" to com.google.android.cameraview.Constants.FLASH_TORCH
+      ),
+      "AutoFocus" to mapOf(
+        "on" to true,
+        "off" to false
+      ),
+      "WhiteBalance" to mapOf(
+        "auto" to com.google.android.cameraview.Constants.WB_AUTO,
+        "cloudy" to com.google.android.cameraview.Constants.WB_CLOUDY,
+        "sunny" to com.google.android.cameraview.Constants.WB_SUNNY,
+        "shadow" to com.google.android.cameraview.Constants.WB_SHADOW,
+        "fluorescent" to com.google.android.cameraview.Constants.WB_FLUORESCENT,
+        "incandescent" to com.google.android.cameraview.Constants.WB_INCANDESCENT
+      ),
+      "VideoQuality" to mapOf(
+        "2160p" to VIDEO_2160P,
+        "1080p" to VIDEO_1080P,
+        "720p" to VIDEO_720P,
+        "480p" to VIDEO_480P,
+        "4:3" to VIDEO_4x3
+      ),
+      "FaceDetection" to emptyMap<String, Any>(),
+    )
 
-import java.lang.Exception
+    AsyncFunction("pausePreview") { viewTag: Int ->
+      val view = findView(viewTag)
 
-class CameraModule(
-  context: Context,
-  private val moduleRegistryDelegate: ModuleRegistryDelegate = ModuleRegistryDelegate()
-) : ExportedModule(context) {
+      if (view.isCameraOpened) {
```

Comment:
No need to change it in this PR if that's the old property, but it should be `isCameraOpen` 😄 basically "opened" is correct only when used as a verb